### ISSUE TITLE
Endless loop on job activation if job does not fit in max message size

### ISSUE THREAD

**Issue by npepinpe** [zeebe-io/zeebe#4420]
**Describe the bug**

When activating the oldest job in the job state would result in a `JobBatchRecord` which is too large to fit in the max message size, we currently end up in an endless loop - the gateway sees that `truncated` property of the response is true, and will retry that partition over and over again, but will never get anything while the job exists, as jobs are deterministically ordered.

Not only does this brick a gateway, but it's also not obvious what is going on at all, making this hard to react to.

**To Reproduce**

<details><summary>Integration Test</summary>

```java
package io.zeebe.broker.it.network;

import static org.assertj.core.api.Assertions.assertThat;

import com.fasterxml.jackson.databind.ObjectMapper;
import io.zeebe.broker.it.util.GrpcClientRule;
import io.zeebe.broker.test.EmbeddedBrokerRule;
import io.zeebe.model.bpmn.Bpmn;
import io.zeebe.model.bpmn.BpmnModelInstance;
import io.zeebe.test.util.BrokerClassRuleHelper;
import java.util.Map;
import java.util.concurrent.TimeUnit;
import org.junit.Before;
import org.junit.ClassRule;
import org.junit.Rule;
import org.junit.Test;
import org.junit.rules.RuleChain;
import org.springframework.util.unit.DataSize;

public final class SmallMessageSizeTest {
  private static final ObjectMapper MAPPER = new ObjectMapper();
  private static final DataSize MAX_MESSAGE_SIZE = DataSize.ofKilobytes(4);
  private static final String LARGE_TEXT = "x".repeat((int) (MAX_MESSAGE_SIZE.toBytes() / 4));

  private static final EmbeddedBrokerRule BROKER_RULE =
      new EmbeddedBrokerRule(b -> b.getNetwork().setMaxMessageSize(MAX_MESSAGE_SIZE));
  private static final GrpcClientRule CLIENT_RULE = new GrpcClientRule(BROKER_RULE);

  @ClassRule
  public static RuleChain ruleChain = RuleChain.outerRule(BROKER_RULE).around(CLIENT_RULE);

  @Rule public final BrokerClassRuleHelper helper = new BrokerClassRuleHelper();

  private String jobType;

  private static BpmnModelInstance workflow(final String jobType) {
    return Bpmn.createExecutableProcess("process")
        .startEvent()
        .serviceTask("task", t -> t.zeebeJobType(jobType))
        .endEvent()
        .done();
  }

  @Before
  public void init() {
    jobType = helper.getJobType();
  }

  @Test
  public void shouldDeployLargeWorkflow() {
    // given
    final var workflowKey = CLIENT_RULE.deployWorkflow(workflow(jobType));
    final var workflowInstanceKey = CLIENT_RULE.createWorkflowInstance(workflowKey);

    // when
    for (int i = 0; i < 4; i++) {
      CLIENT_RULE
          .getClient()
          .newSetVariablesCommand(workflowInstanceKey)
          .variables(Map.of(String.valueOf(i), LARGE_TEXT))
          .send()
          .join();
    }

    // then
    final var response =
        CLIENT_RULE
            .getClient()
            .newActivateJobsCommand()
            .jobType(jobType)
            .maxJobsToActivate(1)
            .send()
            .join(10, TimeUnit.SECONDS);
    assertThat(response.getJobs()).hasSize(1);
  }
}
```

</details>

**Expected behavior**

Busting the max message size should not result in an inability to use a partition until the workflow is terminated. In an ideal world, if the job is too large to fit by itself, we could mark it as failed and raise an incident, that way we can keep processing other jobs. I would also be OK with a solution where the issue is clear and easy to diagnose so users can increase the max message size.

**Environment:**
 - OS: Fedora 31
 - Zeebe Version: 0.24.0-SNAPSHOT
 - Configuration: max message size 4kb

**Comment by Zelldon** [comment]
Not the same issue but related to https://github.com/zeebe-io/zeebe/issues/2158

I think we need to somehow define/enforce a limit, such that we can work with that and reject also larger things.

**Comment by npepinpe** [comment]
@Zelldon let's discuss it before you start :slightly_smiling_face:

**Comment by npepinpe** [comment]
Priority was downgraded as it is not related to support anymore.

**Comment by pihme** [comment]
@npepinpe Should this be fixed in the next version or backported to released versions?

**Comment by pihme** [comment]
@npepinpe What is the truncated field used for???

**Comment by menski** [comment]
@pihme the `truncated` flag in the activate job batch response indicates that not all jobs fit into the response (see issue https://github.com/zeebe-io/zeebe/issues/1896). 

tl;dr the gateway asks for 32 jobs, the broker has 44 jobs but because of variables only 15 jobs fit into the response without exceeding the max message size, the response will have the `truncated` flag set, indicating to the gateway to repoll the same broker and not continuing with the next one.
